src/Hadolint/Rules.hs

```diff
@@ -709,7 +710,29 @@ useJsonArgs = instructionRule code severity message check
   where
     code = "DL3025"
     severity = WarningC
-    message = "Use argumens JSON notation for CMD and ENTRYPOINT arguments"
-    check (Cmd (ArgumentsText _ )) = False
-    check (Entrypoint (ArgumentsText _ )) = False
+    message = "Use arguments JSON notation for CMD and ENTRYPOINT arguments"
+    check (Cmd (ArgumentsText _)) = False
+    check (Entrypoint (ArgumentsText _)) = False
     check _ = True
+
+usePipefail :: Rule
+usePipefail = instructionRuleState code severity message check False
+  where
+    code = "DL4006"
+    severity = WarningC
+    message = "Set the SHELL option -o pipefail before RUN with a pipe in it"
+    check _ _ (Shell args) = (argumentsRule hasPipefailOption args, True)
+    check False _ (Run args) = (False, argumentsRule notHasPipes args)
+    check st _ _ = (st, True)
+    notHasPipes script = not (Bash.hasPipes script)
+    hasPipefailOption script =
+        not $
+        null
+            [ True
+            | cmd <- Bash.findCommands script
+            , validShell <- ["/bin/bash", "/bin/zsh", "bash", "zsh"]
```

Comment:
Now I have just realized that Alpine `ash` should also be on the list.

Reply:
alright, I will add it and then merge